src/evaluators/BinaryExpression.js

```diff
@@ -170,8 +177,14 @@ export function computeBinary(
 
   if (lval instanceof AbstractValue || rval instanceof AbstractValue) {
     let type = getPureBinaryOperationResultType(realm, op, lval, rval, lloc, rloc);
-    return realm.createAbstract(new TypesDomain(type), ValuesDomain.topVal, [lval, rval], ([lnode, rnode]) =>
-      t.binaryExpression(op, lnode, rnode)
+    let [hash, args] = hashBinary(op, lval, rval);
+    return realm.createAbstract(
+      new TypesDomain(type),
+      ValuesDomain.topVal,
+      hash,
+      args,
+      ([lnode, rnode]) => t.binaryExpression(op, lnode, rnode),
+      op
```

Comment:
Separate pull requests make sense. I just would have liked to see that straightforward refactoring as the first pull request, to start with the cleanup, instead of first making a dozen files more complicated...